mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java

```diff
@@ -879,6 +879,22 @@ private MarcBibCollection toMarcBibCollection(QueryResult result) {
     return SnapshotDaoUtil.delete(getQueryExecutor(tenantId), snapshotId);
   }
 
+  @Override
+  public Future<Void> cleanRecords(String tenantId) {
+    Promise<Void> promise = Promise.promise();
+    var purgeOldRecordsQuery = DSL.deleteFrom(RECORDS_LB)
+      .where(RECORDS_LB.STATE.eq(RecordState.OLD).and(RECORDS_LB.MATCHED_ID.in(DSL.select(RECORDS_LB.ID).from(RECORDS_LB).where(RECORDS_LB.STATE.eq(RecordState.DELETED)))));
+    var purgeDeletedRecordsQuery = DSL.deleteFrom(RECORDS_LB)
+      .where(RECORDS_LB.STATE.eq(RecordState.DELETED));
+    ReactiveClassicGenericQueryExecutor executor = getQueryExecutor(tenantId);
+    Future.succeededFuture()
+      .compose(ar -> executor.execute(dsl -> purgeOldRecordsQuery))
+      .compose(ar -> executor.execute(dsl -> purgeDeletedRecordsQuery))
```

Comment:
hi @Igor-Gorchakov, I propose to perform records' deletion operations in transaction